aws/elasticache/main.tf

```diff
@@ -44,38 +46,51 @@ resource "aws_elasticache_replication_group" "mod" {
 }
 
 resource "aws_elasticache_parameter_group" "mod" {
-  count = "${var.parameter_group_name != "" ? 0 : 1}"
+  count = "${var.create_parameter_group ? 1 : 0}"
   name = "${local.parameter_group_name}"
   family = "${local.family}"
   description = "${var.name} ${var.env} env ${var.engine} cluster param group"
+
+  lifecycle {
+    create_before_destroy = true
+  }
 }
 
 resource "aws_elasticache_subnet_group" "mod" {
-  name = "${var.name}-${var.env}-${var.engine}-subnet"
-  description = "${var.name}-${var.env}-${var.engine}-subnet"
+  name = "${local.cluster_name}-${var.engine}-subnet"
+  description = "${local.cluster_name}-${var.engine}-subnet"
   subnet_ids = ["${var.subnets}"]
+
+  lifecycle {
+    create_before_destroy = true
+  }
 }
 
 resource "aws_security_group" "sg_for_access_by_sgs" {
-  name = "${var.env}-${var.engine}"
-  description = "${var.env} to ${var.engine}"
+  count = "${var.provide_sg_for_access ? 1 : 0}"
+  name = "${var.name}_${var.env}-${var.engine}"
+  description = "${var.name}_$${var.env} to ${var.engine}"
```

Comment:
extra $ here. Found that because of the planshots :)

Reply:
Ha. Or you could have said thanks for making planshots more usable :)